core/templates/pages/blog-admin-page/blog-admin-page.component.ts

```diff
@@ -0,0 +1,191 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the blog admin page.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { BlogAdminBackendApiService, ConfigPropertiesBackendResponse, ConfigPropertyValues }
+  from 'domain/blog-admin/blog-admin-backend-api.service';
+import { BlogAdminDataService } from 'pages/blog-admin-page/services/blog-admin-data.service';
+import { AdminTaskManagerService } from 'pages/admin-page/services/admin-task-manager.service';
+import { Schema } from 'services/schema-default-value.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface UpdateRoleAction {
+  newRole: string;
+  username: string;
+  isValid: () => boolean;
+}
+
+interface RemoveEditorRole {
+  username: string
+  isValid: () => boolean;
+}
+
+interface FormData {
+  updateRole: UpdateRoleAction;
+  removeEditorRole: RemoveEditorRole;
+}
+
+@Component({
+  selector: 'oppia-blog-admin-page',
+  templateUrl: './blog-admin-page.component.html',
+})
+export class BlogAdminPageComponent implements OnInit {
+  UPDATABLE_ROLES = {};
+  roleToActions;
+  statusMessage: string = '';
+  formData: FormData;
+  configProperties: ConfigPropertiesBackendResponse = {};
+  constructor(
+    private backendApiService: BlogAdminBackendApiService,
+    private blogAdminDataService: BlogAdminDataService,
+    private adminTaskManagerService: AdminTaskManagerService,
+    private windowRef: WindowRef,
+  ) {}
+
+  ngOnInit(): void {
+    this.refreshFormData();
+    this.blogAdminDataService.getDataAsync().then((DataObject) => {
+      this.UPDATABLE_ROLES = DataObject.updatableRoles;
+      this.roleToActions = DataObject.roleToActions;
+    });
+    this.reloadConfigProperties();
+  }
+
+  refreshFormData(): void {
+    this.formData = {
+      updateRole: {
+        newRole: null,
+        username: '',
+        isValid(): boolean {
+          if (this.newRole === 'BLOG_POST_EDITOR') {
+            return Boolean(this.username);
+          } else if (this.newRole === 'BLOG_ADMIN') {
+            return Boolean(this.username);
+          }
+          return false;
+        }
+      },
+      removeEditorRole: {
+        username: '',
+        isValid(): boolean {
+          if (this.username === '') {
+            return false;
+          }
+          return true;
+        }
+      }
+    };
```

Comment:
IF this directive is in angular Can we use the angular's form module? It provides a better structure for creating such forms structure with a validation method. [Optional or can be done in a separate PR and should not block this PR]

Reply:
Should I create an issue and add a TODO for all the 2 comments together?  Moving `oppia-roles-and-actions-visualizer`, reusing the new roles editor directive here and change this into forms module?

Reply:
Yep let's have one issue with all minor/improvement tasks which you can work on after this PR is merged!

Reply:
https://github.com/oppia/oppia/issues/13327 opened the issue.

Reply:
Thanks!